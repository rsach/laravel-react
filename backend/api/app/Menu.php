<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Menu extends Model
{
    protected $fillable = ['name', 'price', 'picture'];
    //
    public function order_items() {
        return $this->hasMany(Order_items::class);
    }
}
