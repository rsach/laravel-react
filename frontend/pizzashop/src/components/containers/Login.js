import { connect } from 'react-redux'
import { login} from '../../actions'


import Login from "../ui/Login";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onLogin(body){
            dispatch(
                login(body)
            )
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Login)
