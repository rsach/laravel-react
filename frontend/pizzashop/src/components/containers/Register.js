import { connect } from 'react-redux'
import { register} from '../../actions'


import Register from "../ui/Register";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onSignUp(body){
            console.log(body)
            dispatch(
                register(body)
            )

            return body
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Register)
