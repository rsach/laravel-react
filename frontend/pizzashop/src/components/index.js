
import ShowErrors from './containers/ShowErrors'

import '../stylesheets/index.scss'

import React from 'react'

export const App = ({children}) =>
	<div className="app">
		<ShowErrors />
		{children}

	</div>;



export const Whoops404 = ({ location }) => 
	<div className="whoops-404">
		<h1>Whoops , route not found</h1>
		<p>Cannot find content for {location.pathname} </p>
		
	</div>;