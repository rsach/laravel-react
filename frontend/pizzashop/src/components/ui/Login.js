import React, {Component} from 'react';
import PropTypes from "prop-types";
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
// import AppBar from 'material-ui/AppBar';
import AppBar from '@material-ui/core/AppBar';



import history from "../../history";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";

class Login extends Component {
    constructor(props) {
        super(props);

        // reset login status


        this.state = {
            email: '',
            password: '',
            submitted: false
        };


        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }




    handleSubmit(e) {
        e.preventDefault();

        this.setState({ submitted: true });
        const { email, password } = this.state;
        const {onLogin=f=>f} = this.props;

        if (email && password) {

            onLogin({email, password});
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/register')
    }

    render() {


        return (

            <div>
                <MuiThemeProvider>
                    <div>
                        <AppBar
                            title="Login"
                        />
                        <div className="flex  flex-center">
                            <div className="form flex flex-column flex-center">


                                <TextField
                                    id="email"
                                    data-testid="email"
                                    variant="outlined"
                                    placeholder="Enter your Email"
                                    label="Email"
                                    onChange = {(event) => this.setState({email:event.target.value})}
                                />
                                <br/>
                                <TextField
                                    data-testid="password"

                                    id="password"
                                    variant="outlined"
                                    type="password"
                                    placeholder="Enter your Password"
                                    label="Password"
                                    onChange = {(event) => this.setState({password:event.target.value})}
                                />
                                <br/>
                                <div className="flex button-margin flex-space-between">
                                    <Button                    id="login"                 data-testid="login"
                                                                                variant="contained" color="primary"  onClick={(event) => this.handleSubmit(event)}>Login</Button>

                                        <Button id="register" variant="contained"        data-testid="register"
                                                onClick={(e) => this.handleRedirect(e)}  color="secondary"    >
                                            Register
                                        </Button>




                                </div>
                            </div>

                        </div>

                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
}


Login.propTypes = {
    onLogin: PropTypes.func.isRequired
};

export default Login

