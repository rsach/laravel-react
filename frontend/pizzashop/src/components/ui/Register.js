import React, {Component} from 'react';

import PropTypes from "prop-types";


import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';


import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import history from "../../history";


class Register extends Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                name: '',
                email: '',
                password: ''
            },
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }

    handleChange(newValue) {
        // const { name, value } = event;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                ...newValue
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;
        const {onSignUp=f=>f} = this.props;

        if (user.name && user.email && user.password) {

            onSignUp(user);
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/login')
    }

    render() {

        // const { user, submitted } = this.state;
        return (
            <div>
                <MuiThemeProvider>
                <div>
                    <AppBar
                        title="Register"
                    />
                    <div className="flex  flex-center">
                        <div className="form flex flex-column">

                                <TextField
                                    id="name"
                                    variant="outlined"

                                placeholder="Enter your Name"
                                label="Name"
                                onChange = {(event) => this.handleChange({name: event.target.value})}
                            />


                            <br/>
                            <TextField
                                id="email"
                                placeholder="Enter your Email"
                                type="email"
                                variant="outlined"

                                label="Email"
                                onChange = {(event) => this.handleChange({email: event.target.value})}
                            />
                            <br/>
                            <TextField
                                id="password"
                                type = "password"
                                variant="outlined"

                                placeholder="Enter your Password"
                                label="Password"
                                onChange = {(event) => this.handleChange({password: event.target.value})}
                            />
                            <br/>
                            <div className="flex button-margin flex-space-between">
                                <Button id="register" label="Register" variant="contained"  color="primary"  onClick={(event) => this.handleSubmit(event)}> Register </Button>


                                    <Button id="login" variant="contained" label="Login" color="secondary" onClick={(e) => this.handleRedirect(e)}    >

                                        Login
                                    </Button>



                            </div>
                        </div>
                    </div>
                </div>
            </MuiThemeProvider>
            </div>
        );
    }
}


Register.propTypes = {
    onSignUp: PropTypes.func,
};

export default Register
