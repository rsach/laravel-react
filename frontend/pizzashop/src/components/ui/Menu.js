import { Link } from 'react-router-dom'
import HomeIcon from 'react-icons/lib/fa/home'

import {FaShoppingCart, FaArchive} from 'react-icons/lib/fa'
import React from 'react'

import '../../stylesheets/Menu.scss'

const Menu = () => 
		<nav className = "menu">
			<Link to="/" >
				<HomeIcon />
			</Link>
			<Link  to="/checkout">
				<FaShoppingCart />

			</Link>
			<Link  to="/pastOrders">
				<FaArchive />

			</Link>




		</nav>;

export default Menu