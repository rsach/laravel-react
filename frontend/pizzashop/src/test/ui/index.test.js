import React from 'react';
import { render } from '@testing-library/react';
import Login from "../../components/ui/Login";
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Register from "../../components/ui/Register";
import SuccessOrder from "../../components/ui/SuccessOrderPage";
import Menu from "../../components/ui/Menu";
import {App, Whoops404} from "../../components";

Enzyme.configure({ adapter: new Adapter() });


describe('<App/> spec', () => {


    function setup() {
        const props = {
            children: 'test'
        };
        const wrapper = shallow(<App  />);
        return {
            props,  wrapper
        }
    }



    it('renders the component', () => {
        const {wrapper} = setup();
        expect(wrapper.firstChild).toMatchSnapshot();
    });



});


describe('<Whoops404/> spec', () => {

    // const { location } = window;
    //
    // beforeAll(() => {
    //     delete window.location;
    //     window.location = { reload: 'test' };
    //
    // });
    //
    // afterAll(() => {
    //     window.location = location;
    // });
    function setup() {
        const props = {

            location: {pathname: 'test'}
        };
        const wrapper = shallow(<Whoops404 {...props} />);
        return {
            props,  wrapper
        }
    }



    it('renders the component', () => {
        const {wrapper} = setup();
        expect(wrapper.firstChild).toMatchSnapshot();
    });



});