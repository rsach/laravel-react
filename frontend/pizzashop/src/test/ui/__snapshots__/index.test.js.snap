// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> spec renders the component 1`] = `undefined`;

exports[`<Whoops404/> spec renders the component 1`] = `undefined`;
